File: solutions/python/dnd-character/1/dnd_character.py
import random

def roll_dice():
    rolls = []
    for i in range(4):
        rolls.append(random.randint(1, 6))
    return sorted(rolls)

class Character:
    def __init__(self):
        self.strength = sum(roll_dice()[1:])
        self.dexterity = sum(roll_dice()[1:])
        self.constitution = sum(roll_dice()[1:])
        self.intelligence = sum(roll_dice()[1:])
        self.wisdom = sum(roll_dice()[1:])
        self.charisma = sum(roll_dice()[1:])

        self.hitpoints = 10 + modifier(self.constitution)

    def ability(self):
        return self.constitution

def modifier(value):
    return (value - 10) // 2

File: solutions/csharp/dnd-character/1/DndCharacter.cs
public class DndCharacter
{
  private static readonly Random _random = new();

  public int Strength { get; }
  public int Dexterity { get; }
  public int Constitution { get; }
  public int Intelligence { get; }
  public int Wisdom { get; }
  public int Charisma { get; }
  public int Hitpoints { get; }

  public DndCharacter()
  {
    Strength = Ability();
    Dexterity = Ability();
    Constitution = Ability();
    Intelligence = Ability();
    Wisdom = Ability();
    Charisma = Ability();

    Hitpoints = 10 + Modifier(Constitution);
  }

  public static int Modifier(int score)
  {
    return (int)Math.Floor((score - 10) / 2.0);
  }

  public static int Ability()
  {
    return RollFourD6().Sum();
  }

  public static DndCharacter Generate()
  {
    return new DndCharacter();
  }

  private static List<int> RollFourD6()
  {
    return Enumerable.Range(1, 4)
      .Select(_ => _random.Next(1, 7))
      .OrderByDescending(n => n)
      .Take(3)
      .ToList();
  }
}
